### ISSUE TITLE
openSUSE won't build with desktop categories Categories=Audio;X-lsp-plugins;

### ISSUE THREAD

**Issue by kevinsmia1939** [sadko4u/lsp-plugins#97]
Hello,

I could not build lsp-plugins 1.1.17 on openSUSE Tumbleweed.
```
ERROR: No sufficient Category definition "Audio;X-lsp-plugins;" in /home/abuild/rpmbuild/BUILDROOT/lsp-plugins-1.1.17-0.x86_64//usr/share/applications/XXXX.desktop 
Please refer to https://en.opensuse.org/openSUSE:Packaging_desktop_menu_categories
```

I think X-lsp-plugins category is not recognized by openSUSE. The desktop file in /res/xdg might have to be change.

Try using these instead.

```
    AudioVideoEditing
    HamRadio
    Midi
    Mixer
    Player
    Recorder
    Sequencer
    Tuner
```

from https://en.opensuse.org/openSUSE:Packaging_desktop_menu_categories

Thanks.

**Comment by sadko4u** [comment]
Hello!
Please consider looking at the discussion here: #88
Do you have any suggestions how to fix this and keep 'LSP Plugins' submenu in the 'Audio' menu?

**Comment by sadko4u** [comment]
@eeickmeyer @dvzrv 
I'm tired of continuous XDG-related issues in Github tracker.
If things will last so, I'll remove *.desktop icons from the installation script until we get a proper solution.

**Comment by sadko4u** [comment]
I need to test how the system will behave if I remove 'X-lsp-plugins' from categories.

**Comment by eeickmeyer** [comment]
> why not do it as upstream documentation suggests and pull in the the desktop files via a properly defined menu item?

Because I didn't think of it??? :joy:

**Comment by sadko4u** [comment]
> The lsp-plugins applications are standalone jack clients

Moreover, they're much easier to debug when they run as standalone JACK applications.

**Comment by sadko4u** [comment]
Removed x-lsp-plugins from *.desktop files. Let's try: 7c9f2bc473c9e291cf1edc0146b3900bd94915c4

**Comment by dvzrv** [comment]
@sadko4u I removed the entire `Categories` from the .desktop files. If you leave e.g. Audio in the Categories the desktop files will show up in any "Audio" menu *and* in the assigned menu.
In XFCE the desktop entries show properly in the menu: `Multimedia -> LSP Plugins`

**Comment by eeickmeyer** [comment]
> @sadko4u I removed the entire `Categories` from the .desktop files. If you leave e.g. Audio in the Categories the desktop files will show up in any "Audio" menu and in the assigned menu.
In XFCE the desktop entries show properly in the menu: Multimedia -> LSP Plugins

This is likely entirely the right way to do this.

**Comment by sadko4u** [comment]
In KDE I've got a menu item and each plugin as a submenu element.

**Comment by sadko4u** [comment]
Also please remember: X.Org may cache menu items and icons. To properly view the result, you need to restart X.Org. Logout/Login may not help.

**Comment by sadko4u** [comment]
LXDE works just fine.

**Comment by sadko4u** [comment]
Gnome under openSUSE shows plugins and, moreove, icons for them.

**Comment by sadko4u** [comment]
IceWM shows in both Multimedia and LSP Plugins submenu.

**Comment by dvzrv** [comment]
@sadko4u however, it might be that the various DEs behave differently when encountering a desktop file that is both pulled in by a menu *and* defines a `Categories` keyword.

As mentioned before, I have removed the entire `Categories` keyword in all desktop files.
I've tested this in KDE and XFCE now. They all show the `LSP Plugins` menu with the plugin desktop files in them correctly and none are duplicated in other locations.

Gnome doesn't have a menu anymore by default. Activating it in the tweaks application shows the icons in `Sound & Video` for whatever reason. The menu in Gnome I'd not consider standard though, as it doesn't follow XDG standards (in regards to defined menus and their items).

**Comment by sadko4u** [comment]
Gnome in Ubuntu doesn't show icons. What's wrong with Ubuntu?

**Comment by dvzrv** [comment]
Not entirely sure. Gnome is sadly often weird (it showed the svg graphics for me on Arch Linux though).

I have also tested on LXDE and it works just fine when removing the `Categories` entirely.
I think this will be the best solution! :)

**Comment by sadko4u** [comment]
Can confirm that removing categories works just fine with KDE Plasma 5.

**Comment by eeickmeyer** [comment]
> What's wrong with Ubuntu?

It's entirely a Gnome issue. They have chosen not to follow XDG standards. There's a reason Ubuntu Studio doesn't use it. :)

**Comment by sadko4u** [comment]
> > What's wrong with Ubuntu?
> 
> It's entirely a Gnome issue. They have chosen not to follow XDG standards. There's a reason Ubuntu Studio doesn't use it. :)

But GNOME in openSUSE Tumbleweed just works fine!
Here's openSUSE:
![image](https://user-images.githubusercontent.com/12834464/78607361-97163100-7867-11ea-8fd7-45566d32da6d.png)

And here's Ubuntu:
![image](https://user-images.githubusercontent.com/12834464/78607400-a7c6a700-7867-11ea-81a7-1ecc2f1d5e10.png)

**Comment by eeickmeyer** [comment]
Tumbleweed is much faster at implementing upstream bugfixes than Ubuntu.

**Comment by sadko4u** [comment]
Just updated packages in Ubuntu and it got wokring.
![image](https://user-images.githubusercontent.com/12834464/78608306-58817600-7869-11ea-83a6-a95a31a317d6.png)

**Comment by dvzrv** [comment]
All in all I'm happy it's otherwise working all across the board actually :)

**Comment by sadko4u** [comment]
@eeickmeyer 
I've installed rare Ubuntu on my virtual machine because had to reproduce https://github.com/KXStudio/Repository/issues/219 issue.
Now I'm using this virtual machine for tests, too :).

**Comment by eeickmeyer** [comment]
`Just updated packages in Ubuntu and it got wokring.`
Then I happily stand corrected.

**Comment by sadko4u** [comment]
Merged changs into **devel** branch: 2cb434dbe551320e30a977da04bbb653323a1905

**Comment by sadko4u** [comment]
Changes are available in **hotfix** branch as a single patch: bd357bf0718a58dab3ec655d344282e938067080

**Comment by dvzrv** [comment]
On 2020-04-06 15:09:39 (-0700), Vladimir Sadovnikov wrote:
> Changes are available in **hotfix** branch as a single patch: bd357bf0718a58dab3ec655d344282e938067080

Awesome stuff. Thank you!

**Comment by sadko4u** [comment]
Removed **github-issue-97** branch.

**Comment by kevinsmia1939** [comment]
I will ask openSUSE community about this, thanks for the update.

**Comment by kevinsmia1939** [comment]
I ask openSUSE community, they said xdg standard does not specify category. But openSUSE is probably the only distro to check those category.

**Comment by hellcp** [comment]
> Hello!
> Please consider looking at the discussion here: #88
> Do you have any suggestions how to fix this and keep 'LSP Plugins' submenu in the 'Audio' menu?

Very simple, ship a [menu extension](https://specifications.freedesktop.org/menu-spec/menu-spec-latest.html#third-party-howto) file with a directory file instead of inventing a new category.

> @sadko4u Looks like the openSUSE community has a disagreement about the X-* subcategories. XDG allows them

I will need a quote in the XDG spec for that, because I see no such thing in any of them.

**Comment by sadko4u** [comment]
@hellcp 
> Very simple, ship a menu extension file with a directory file instead of inventing a new category.

Yes, that's done in that way currently.

> I will need a quote in the XDG spec for that, because I see no such thing in any of them.

[Extensions to the desktop entry formats](https://specifications.freedesktop.org/menu-spec/menu-spec-latest.html#desktop-entry-extensions)

The first paragraph:
>The Categories field is a list of strings used to classify menu items. For example, applications in the AudioVideo category might end up in the "Sound & Video" submenu. Appendix A, Registered Categories enumerates the standard categories. **Categories not in this document must be prefixed by the string "X-" indicating that they are extensions**. Categories are case-sensitive.

**Comment by sadko4u** [comment]
@kevinsmia1939 

Could you please ship a link to the topic in the openSUSE community?

**Comment by hellcp** [comment]
> > I will need a quote in the XDG spec for that, because I see no such thing in any of them.
> 
> [Extensions to the desktop entry formats](https://specifications.freedesktop.org/menu-spec/menu-spec-latest.html#desktop-entry-extensions)
> 
> The first paragraph:
> 
> > The Categories field is a list of strings used to classify menu items. For example, applications in the AudioVideo category might end up in the "Sound & Video" submenu. Appendix A, Registered Categories enumerates the standard categories. **Categories not in this document must be prefixed by the string "X-" indicating that they are extensions**. Categories are case-sensitive.

Ha, I missed that, I will have to find a way to fix our checks then, thanks for that

**Comment by hellcp** [comment]
As an aside @kevinsmia1939 you should use [%suse_update_desktop_file](https://en.opensuse.org/openSUSE:Packaging_Conventions_RPM_Macros#.25suse_update_desktop_file) to add enough categories so it doesn't complain there aren't enough of them for now, while I think how to write the check in a way that doesn't break everything ;)

**Comment by kevinsmia1939** [comment]
> @kevinsmia1939
> 
> Could you please ship a link to the topic in the openSUSE community?

@hellcp is the person you can talk about openSUSE.

**Comment by sadko4u** [comment]
Available in 1.1.19 release!
